### PR TITLE
Build m32 from upstream

### PR DESCRIPTION
**Pull request by gretay-js**
Follow-up on #387. Provide a way to build and install targets from ocaml/ subdirectory. Can be particularly useful for targets not available in backend subdirectory. Tested for  i386.

This PR adds "make install_stage0" target that installs the result of stage0 build to the configured prefix.

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -228,6 +228,13 @@ config_files: ocaml-stage1-config.status
 	/bin/echo -n "( `grep \"^SHAREDLIB_CFLAGS=\" ocaml/Makefile.config \
 		| sed 's/^SHAREDLIB_CFLAGS=//'` )" > sharedlib_cflags.sexp
 
+## For targets not supported in backend, we can install the result of stage0
+.PHONY: install_stage0
+install_stage0: stage0
+	rsync --copy-links --chmod=u+rw,go+r -r $(stage0_prefix)/* $(prefix)
+        # CR gyorsh: when is it supposed to be installed for upstream build?
```
Comment (mshinwell) [inline Makefile.in:235]: I don't know the answer to this question -- presumably just install it at the same time as we do at the moment?

File: Makefile.in [hunk 2]
```diff
@@ -228,6 +228,13 @@ config_files: ocaml-stage1-config.status
 	/bin/echo -n "( `grep \"^SHAREDLIB_CFLAGS=\" ocaml/Makefile.config \
 		| sed 's/^SHAREDLIB_CFLAGS=//'` )" > sharedlib_cflags.sexp
 
+## For targets not supported in backend, we can install the result of stage0
+.PHONY: install_stage0
+install_stage0: stage0
+	rsync --copy-links --chmod=u+rw,go+r -r $(stage0_prefix)/* $(prefix)
```
Comment (mshinwell) [inline Makefile.in:234]: I think before this `rsync` line we need a `mkdir -p` of the prefix directory (see the usual installation rules elsewhere in this file).

**Review by mshinwell — CHANGES_REQUESTED** [review]

File: Makefile.in (hunk 2 above)
Comment (gretay-js) [reply Makefile.in:235]: Thank you for spotting this! Fixed.

File: Makefile.in (hunk 1 above)
Comment (gretay-js) [reply Makefile.in]: I removed the CR in the latest commit.

File: Makefile.in [hunk 3]
```diff
@@ -228,6 +228,28 @@ config_files: ocaml-stage1-config.status
...
+	rm -rf _build_upstream
+	mkdir _build_upstream
+	rsync -a $$(pwd)/ocaml/ $$(pwd)/_build_upstream
+	(cd _build_upstream && \
+	  cat ../configure_opts_stage0 | tr -d '\n' | xargs -0 ./configure -C \
+	    --prefix=@prefix@ \
+	    --disable-ocamldoc \
+	    --disable-ocamltest \
+	    --disable-debug-runtime \
+	    --disable-instrumented-runtime \
```
Comment (stedolan) [inline Makefile.in:243]: Looks good, but would you mind re-enabling the debug/instrumented runtime?

**Review by mshinwell — APPROVED** [review]

File: Makefile.in (hunk 3 above)
Comment (gretay-js) [reply Makefile.in]: Done.

**Review by poechsel — APPROVED** [review]

**Pull request merged**